hedera-node/src/main/java/com/hedera/services/queries/token/GetAccountNftInfoAnswer.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    @Override
+    public ResponseCodeEnum checkValidity(Query query, StateView view) {
+        var accountNftInfoQuery = query.getTokenGetAccountNftInfo();
+        var validity = validator.nftMaxQueryRangeCheck(accountNftInfoQuery.getStart(), accountNftInfoQuery.getEnd());
+        if (validity != OK) {
+            return validity;
+        }
+
+        var accountID = accountNftInfoQuery.getAccountID();
+
+        return view.accounts().containsKey(accountID) ? OK : INVALID_ACCOUNT_ID;
+    }
+
+    @Override
+    public HederaFunctionality canonicalFunction() {
+        return HederaFunctionality.TokenGetAccountNftInfo;
+    }
+
+    @Override
+    public ResponseCodeEnum extractValidityFrom(Response response) {
+        return response.getTokenGetAccountNftInfo().getHeader().getNodeTransactionPrecheckCode();
+    }
+
+    @Override
+    public Optional<SignedTxnAccessor> extractPaymentFrom(Query query) {
+        var paymentTxn = query.getTokenGetAccountNftInfo().getHeader().getPayment();
+        return Optional.ofNullable(uncheckedFrom(paymentTxn));
+    }
+
+    private Response responseFor(
+            Query query,
+            StateView view,
+            ResponseCodeEnum validity,
+            long cost,
+            Optional<Map<String, Object>> queryCtx
+    ) {
+        var op = query.getTokenGetAccountNftInfo();
+        var response = TokenGetAccountNftInfoResponse.newBuilder();
+
+        var type = op.getHeader().getResponseType();
+        if (validity != OK) {
+            response.setHeader(header(validity, type, cost));
+        } else {
+            if (type == COST_ANSWER) {
+                response.setHeader(costAnswerHeader(OK, cost));
+            } else {
+                setAnswerOnly(response, view, op, cost, queryCtx);
+            }
+        }
+
+        return Response.newBuilder()
+                .setTokenGetAccountNftInfo(response)
+                .build();
+    }
+
+    @SuppressWarnings("unchecked")
+    private void setAnswerOnly(
+            TokenGetAccountNftInfoResponse.Builder response,
+            StateView view,
+            TokenGetAccountNftInfoQuery op,
+            long cost,
+            Optional<Map<String, Object>> queryCtx
+    ) {
+        if (queryCtx.isPresent()) {
+            var ctx = queryCtx.get();
+            if (!ctx.containsKey(ACCOUNT_NFT_INFO_CTX_KEY)) {
+                response.setHeader(answerOnlyHeader(INVALID_ACCOUNT_ID));
+            } else {
+                response.setHeader(answerOnlyHeader(OK, cost));
+                response.addNfts((TokenNftInfo) ctx.get(ACCOUNT_NFT_INFO_CTX_KEY));
+            }
+        } else {
+            var info = view.infoForAccountNft(op.getAccountID(), op.getStart(), op.getEnd());
+            if (info.isEmpty()) {
+                response.setHeader(answerOnlyHeader(INVALID_NFT_ID));
+            } else {
+                response.setHeader(answerOnlyHeader(OK, cost));
+                response.addAllNfts(info.get());
```

Comment:
I think that the proto is different from the HIP description:
https://github.com/hashgraph/hedera-improvement-proposal/blob/master/HIP/hip-17.md#tokengetnftinfo